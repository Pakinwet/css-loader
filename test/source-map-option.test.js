import path from 'path';

import webpack from './helpers/compiler';
import evaluated from './helpers/evaluated';
import { normalizeModule } from './helpers/utils';

describe('sourceMap', () => {
  const generateRulesWithSourceMap = (enableSourceMap, sourceMap) => {
    return {
      rules: [
        {
          test: /\.css$/,
          use: [
            {
              loader: path.resolve(__dirname, '../src'),
              options: {
                sourceMap: enableSourceMap,
              },
            },
            {
              loader: path.resolve(__dirname, 'fixtures/source-map-loader.js'),
              options: {
                sourceMap,
              },
            },
          ],
        },
      ],
    };
  };

  describe('true', () => {
    test('basic', async () => {
      const config = {
        loader: {
          options: {
            sourceMap: true,
          },
        },
      };
      const stats = await webpack('source-map/basic.css', config);
      const { modules } = stats.toJson();

      expect(
        normalizeModule(evaluated(modules[modules.length - 1].source))
      ).toMatchSnapshot('module');

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    test('map from other loader', async () => {
      const config = generateRulesWithSourceMap(true, {
        file: 'test.css',
        mappings: 'AAAA,SAAS,SAAS,EAAE',
        names: [],
        sourceRoot: '',
        sources: ['/folder/test.css'],
        sourcesContent: ['.class { a: b c d; }'],
        version: 3,
      });
      const stats = await webpack('source-map/basic.css', config);
      const { modules } = stats.toJson();

      expect(
        normalizeModule(evaluated(modules[modules.length - 1].source))
      ).toMatchSnapshot('module');

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    test('map is string', async () => {
      const config = generateRulesWithSourceMap(
        true,
        JSON.stringify({
          file: 'test.css',
          mappings: 'AAAA,SAAS,SAAS,EAAE',
          names: [],
          sourceRoot: '',
          sources: ['/folder/test.css'],
          sourcesContent: ['.class { a: b c d; }'],
          version: 3,
        })
      );
      const stats = await webpack('source-map/basic.css', config);
      const { modules } = stats.toJson();

      expect(
        normalizeModule(evaluated(modules[modules.length - 1].source))
      ).toMatchSnapshot('module');

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });
  });

  describe('false', () => {
    test('basic', async () => {
      const config = {
        loader: {
          options: {
            sourceMap: false,
          },
        },
      };
      const stats = await webpack('source-map/basic.css', config);
      const { modules } = stats.toJson();

      expect(evaluated(modules[modules.length - 1].source)).toMatchSnapshot(
        'module'
      );

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });

    test('map from other loader', async () => {
      const config = generateRulesWithSourceMap(false, {
        file: 'test.css',
        mappings: 'AAAA,SAAS,SAAS,EAAE',
        names: [],
        sourceRoot: '',
        sources: ['/folder/test.css'],
        sourcesContent: ['.class { a: b c d; }'],
        version: 3,
      });
      const stats = await webpack('source-map/basic.css', config);
      const { modules } = stats.toJson();

      expect(evaluated(modules[modules.length - 1].source)).toMatchSnapshot(
        'module'
      );

      expect(stats.compilation.warnings).toMatchSnapshot('warnings');
      expect(stats.compilation.errors).toMatchSnapshot('errors');
    });
  });
});
