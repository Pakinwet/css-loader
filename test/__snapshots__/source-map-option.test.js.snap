// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sourceMap false basic: errors 1`] = `Array []`;

exports[`sourceMap false basic: module 1`] = `
Array [
  Array [
    1,
    ".class {
  color: red;
}
",
    "",
  ],
]
`;

exports[`sourceMap false basic: warnings 1`] = `Array []`;

exports[`sourceMap false map from other loader: errors 1`] = `Array []`;

exports[`sourceMap false map from other loader: module 1`] = `
Array [
  Array [
    1,
    ".class {
  color: red;
}
",
    "",
  ],
]
`;

exports[`sourceMap false map from other loader: warnings 1`] = `Array []`;

exports[`sourceMap true basic: errors 1`] = `Array []`;

exports[`sourceMap true basic: module 1`] = `
Array [
  Array [
    1,
    ".class {
  color: red;
}
",
    "",
    Object {
      "file": "basic.css",
      "mappings": "AAAA;EACE,WAAW;CACZ",
      "names": Array [],
      "sourceRoot": "",
      "sources": Array [
        "/replaced/original/path/",
      ],
      "sourcesContent": Array [
        ".class {
  color: red;
}
",
      ],
      "version": 3,
    },
  ],
]
`;

exports[`sourceMap true basic: warnings 1`] = `Array []`;

exports[`sourceMap true map from other loader: errors 1`] = `Array []`;

exports[`sourceMap true map from other loader: module 1`] = `
Array [
  Array [
    1,
    ".class {
  color: red;
}
",
    "",
    Object {
      "file": "basic.css",
      "mappings": "AAAA;ECCE,WAAW;CACZ",
      "names": Array [],
      "sourceRoot": "",
      "sources": Array [
        "/replaced/original/path/",
        "/replaced/original/path/",
      ],
      "sourcesContent": Array [
        ".class { a: b c d; }",
        ".class {
  color: red;
}
",
      ],
      "version": 3,
    },
  ],
]
`;

exports[`sourceMap true map from other loader: warnings 1`] = `Array []`;

exports[`sourceMap true map is string: errors 1`] = `Array []`;

exports[`sourceMap true map is string: module 1`] = `
Array [
  Array [
    1,
    ".class {
  color: red;
}
",
    "",
    Object {
      "file": "basic.css",
      "mappings": "AAAA;ECCE,WAAW;CACZ",
      "names": Array [],
      "sourceRoot": "",
      "sources": Array [
        "/replaced/original/path/",
        "/replaced/original/path/",
      ],
      "sourcesContent": Array [
        ".class { a: b c d; }",
        ".class {
  color: red;
}
",
      ],
      "version": 3,
    },
  ],
]
`;

exports[`sourceMap true map is string: warnings 1`] = `Array []`;
